### PR TITLE
Document CI setup

### PR DESCRIPTION
**Pull request by MaggieSalak**
# PR for issue #1015 

## What is being addressed

Documentation of CI setup was missing.

## How is this addressed

New documentation page created describing what workflows are in use, what tasks they execute and what settings they have. Additional execution details were added for the E2E CI.

### CODE REVIEW

File: docs/user-guide/ci-setup.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+- Builds and pushes a docker image for the decoder
+
+## Other workflows
+
+The repository contains other workflows which are run automatically under
+certain confitions, typically when a pull request is created:
+
+- [CodeQL][codeql] - runs CodeQL analysis on each PR created against `dev`
+  branch and after merge
+- [Lint and Check Markdown][lint-markdown] - runs linter and checks links in
```
Comment (roel4ez) [inline docs/user-guide/ci-setup.md:82]: This one only runs on `docs/main`

File: docs/user-guide/ci-setup.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  markdown files; runs on PR created agains `dev` branch if the PR contains
+  `.md` files
+- [Test Report][test-report] - publishes test results after Build & Test CI or
+  E2E CI have completed
+- [Publish docs][publish-docs] - publishes project documentation using MkDocs.
```
Comment (roel4ez) [inline docs/user-guide/ci-setup.md:87]: There are two different ones: 
1. runs on changes to `docs/main` and publishes `dev` version of docs
2. only runs manually and requires a version as input to run. publishes that specific version.

Comment (MaggieSalak) [reply docs/user-guide/ci-setup.md]: Good point, I corrected it.

**Review by roel4ez — APPROVED** [review]
LGTM

**Pull request merged**